libs/java/cert_refresher/src/main/java/com/oath/auth/OpenTelemetryCertReloadEventEmitter.java

```diff
@@ -0,0 +1,153 @@
+
+    private static final String GAUGE_RESULT_TIMESTAMP_NAME = "athenz_cert_refresher.refresh.result_last_timestamp";
+    private static final String GAUGE_RESULT_TIMESTAMP_DESC = "Unix timestamp of last refresh result by status (success/failure)";
+
+    private static final String GAUGE_VALIDITY_NAME = "athenz_cert_refresher.service_cert.validity.remaining_secs";
+    private static final String GAUGE_VALIDITY_DESC = "Number of seconds remaining before the current service TLS certificate expires";
+
+    public static final String RESULT_SUCCESS = "success";
+    public static final String RESULT_FAILURE = "failure";
+
+    private final LongCounter refreshResultCounter;
+    private final LongGauge resultLastTimestampGauge;
+    private final LongGauge certValidityRemainingSecs;
+
+    /**
+     * Create OpenTelemetry X.509 cert refresh metrics.
+     *
+     */
+    public OpenTelemetryCertReloadEventEmitter() {
+        Meter meter = GlobalOpenTelemetry.get().getMeter(SCOPE_NAME);
+
+        this.refreshResultCounter = meter.counterBuilder(COUNTER_NAME)
+                .setDescription(COUNTER_DESC)
+                .build();
+
+        this.resultLastTimestampGauge = meter.gaugeBuilder(GAUGE_RESULT_TIMESTAMP_NAME)
+                .setDescription(GAUGE_RESULT_TIMESTAMP_DESC)
+                .ofLongs()
+                .build();
+
+        this.certValidityRemainingSecs = meter.gaugeBuilder(GAUGE_VALIDITY_NAME)
+                .setDescription(GAUGE_VALIDITY_DESC)
+                .ofLongs()
+                .build();
+        
+        LOGGER.info("OpenTelemetry cert refresh metrics initialized");
+    }
+
+    /**
+     * Record a successful certificate refresh (result: success).
+     *
+     * @param certPath path to the certificate file
+     */
+    public void recordCertRefresh(String certPath) {
+        recordRefreshResult(true);
+        exportServiceCertMetric(certPath);
+    }
+
+    /**
+     * Record a failed certificate refresh (result: failure).
+     *
+     * @param errorMessage error description
+     */
+    public void recordCertRefreshFailure(String errorMessage) {
+        recordRefreshResult( false);
+        LOGGER.error("Certificate refresh failed: {}", errorMessage);
+    }
+
+    /**
+     * Record refresh result with counter and timestamp.
+     */
+    private void recordRefreshResult(boolean success) {
+        String result = success ? RESULT_SUCCESS : RESULT_FAILURE;
+        long timestamp = System.currentTimeMillis() / 1000;
+        
+        Attributes attrs = Attributes.builder()
+                .put("function", "cert_refresh")
+                .put("result", result)
+                .build();
+        
+        refreshResultCounter.add(1, attrs);
+        resultLastTimestampGauge.set(timestamp, attrs);
+    }
+
+    /**
+     * Export service certificate validity metric.
+     *
+     * @param certPath path to the certificate file
+     */
+    public void exportServiceCertMetric(String certPath) {
```

Comment:
perhaps we need to rename the method to more neutral "exportCertMetric"? since people role certificates also in addition to service certificates?